yesod-core/src/Yesod/Core/Handler.hs

```diff
@@ -1686,3 +1687,11 @@ getRouteToParent = liftSubHandler $ SubHandlerFor $ return . rheRouteToMaster .
 
 getSubCurrentRoute :: MonadHandler m => m (Maybe (Route (SubHandlerSite m)))
 getSubCurrentRoute = liftSubHandler $ SubHandlerFor $ return . rheRoute . handlerEnv
+
+hasJsonContentType :: MonadHandler m => m (Either String ())
```

Comment:
I think I'd go in the direction of returning a `Bool`, I don't see an advantage to the `Either` return value. Also, when ready, please include a Haddock comment including a `@since` declaration.

Reply:
I went with Either cause parseCheckJsonBody in the error case returns the content-type (you can see it in the changes for this PR). I guess we should not change that. Do you see a better way than using a Either?

Reply:
It's possible at the call site to ask a second time for the mime type. Another option would be a return value such as:

```haskell
data IsJsonMime = JsonMime | NonJsonMime !ByteString
```